packages/flutterfire_ui/lib/src/firestore/table_builder.dart

```diff
@@ -833,16 +865,19 @@ class _Source extends DataTableSource {
             }
           : null,
       cells: [
-        for (final head in getHeaders().keys)
+        for (final head in colList)
           DataCell(
-            _ValueView(data[head]),
-            onTap: () {
-              onEditItem(
-                doc,
-                data[head],
-                head,
-              );
-            },
+            builder?.call(data, colList.indexOf(head)) ??
```

Comment:
I'd rather pass an instance of the head directly instead of passing an index.
Alternatively, we could define a builder on a head instance.

@rrousselGit thoughts?

Reply:
Using an index will allow this kind of logic:

```
(data, colIndex) {
        final person = Person.fromMap(data);
        if (colIndex == 0) {
          return Text(person.firstName);
        }

        if (colIndex == 1) {
          return Row(
            children: [
              Text(person.address.street),
              Text(person.address.city),
            ],
          );
        }
        return Container();
      }
```

I think relying on the header key will tie it to the header implementation (today relying on String key...tomorrow ?).

Reply:
Make sense.  I didn't think of reordering column ! 🙌

Will do the change.

And will also take it into consideration for my rewrite proposal at: https://github.com/firebase/flutterfire/discussions/8569